### ISSUE TITLE
[BUG] Agent v7.40.1 on startup CORE | ERROR | (pkg/collector/worker/check_logger.go:69 in Error) | check:kubelet

### ISSUE THREAD

**Issue by anden-dev** [DataDog/datadog-agent#14272]
**Agent Environment**

Agent v7.40.1

```sh

Defaulted container "agent" out of: agent, trace-agent, process-agent, init-volume (init), init-config (init)
2022-11-10 08:33:15 UTC | CORE | ERROR | (pkg/collector/worker/check_logger.go:69 in Error) | check:kubelet | Error running check: [{"message": "401 Client Error: Unauthorized for url: https://<ENDPOINT-IP>:10250/metrics/cadvisor", "traceback": "Traceback (most recent call last):\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/base.py\", line 1122, in run\n    self.check(instance)\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/kubelet/kubelet.py\", line 387, in check\n    self.process(self.cadvisor_scraper_config, metric_transformers=self.transformers)\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/mixins.py\", line 573, in process\n    for metric in self.scrape_metrics(scraper_config):\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/mixins.py\", line 500, in scrape_metrics\n    response = self.poll(scraper_config)\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/mixins.py\", line 846, in poll\n    response.raise_for_status()\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/requests/models.py\", line 1021, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\nrequests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://<ENDPOINT-IP>:10250/metrics/cadvisor\n"}]
```

Service Check however looks good after this

```sh
 k exec -it datadog-sqsgp -c agent -- agent status | grep kubelet
    kubelet (7.5.0)
      Instance ID: kubelet:5bbc63f3938c02f4 [OK]
````

**Describe what happened:**

During a version upgrade, we discovered error messages in fresh pods 

**Describe what you expected:**

I expected clean logfiles without errors when a node is added. 

**Steps to reproduce the issue:**

Kill a pod of the datadog daemonset. check the logs of the reconsolidated new pod.

**Additional environment details (Operating System, Cloud provider, etc):**
Running on Kubernetes running in AWS EKS nodes are amzn2-ami-hvm-2.0.20221101.0-x86_64-gp2

Related Tickets: 
- https://github.com/DataDog/helm-charts/issues/786
- https://github.com/DataDog/datadog-agent/issues/13883

[DD_Support_Case_Id-1001767](https://help.datadoghq.com/hc/en-us/requests/1001767)

**Comment by vickenty** [comment]
Hi $github_user, thanks for using Datadog.

This issue queue is primarily intended for tracking features, bugs and work items associated with the datadog-agent open source project. The question here is why kubelet rejects the first request from the agent, which is hard to answer by looking at the agent alone, and we may need some more information about your environment. I see you already have a support case open, and it is better to continue this investigation there.

**Comment by anden-dev** [comment]
For the audience fixed with Agent v7.42.0
